packages/core/src/fhirpath/atoms.ts

```diff
@@ -352,6 +352,35 @@ export class XorAtom extends InfixOperatorAtom {
   }
 }
 
+/**
+ * 6.5.5. implies
+ * Returns true if left is true and right is true,
+ * true left is false and right true, false or empty
+ * true left is empty
+ */
+export class ImpliesAtom extends InfixOperatorAtom {
+  constructor(left: Atom, right: Atom) {
+    super('implies', left, right);
+  }
+
+  eval(context: AtomContext, input: TypedValue[]): TypedValue[] {
+    const leftResult = this.left.eval(context, input);
+    const rightResult = this.right.eval(context, input);
+    if (leftResult.length === 0 && rightResult.length === 0) {
+      return [];
+    }
+    const leftValue = leftResult.length === 0 ? null : leftResult[0].value;
```

Comment:
In the case where either argument collection has more than one element in it, this should probably produce an error to let the user know that the expression is incorrect.  In general, FHIRPath is pretty strict when coercing from a collection to a single value; that part of evaluation is covered explicitly here: http://hl7.org/fhirpath/N1/#singleton-evaluation-of-collections